compiler/rustc_passes/src/liveness.rs

```diff
@@ -1686,6 +1710,38 @@ impl<'tcx> Liveness<'_, 'tcx> {
         }
     }
 
+    fn has_added_lit_match_name_span(
+        &self,
+        name: &str,
+        opt_body: Option<&hir::Body<'_>>,
+        err: &mut rustc_errors::DiagnosticBuilder<'_, ()>,
+    ) -> bool {
+        let mut has_litstring = false;
+        if let Some(opt_body) = opt_body {
+            let mut visitor = CollectLitsVisitor { lit_exprs: vec![] };
+            intravisit::walk_body(&mut visitor, opt_body);
+            for lit_expr in visitor.lit_exprs {
+                if let hir::ExprKind::Lit(litx) = &lit_expr.kind {
+                    if let rustc_ast::LitKind::Str(syb, _) = litx.node {
+                        let name_str: &str = syb.as_str();
+                        let mut name_pa = String::from("{");
+                        name_pa.push_str(&name);
+                        name_pa.push('}');
+                        if name_pa.eq(name_str) {
+                            err.span_label(
+                    lit_expr.span,
+                    "you might have meant to use string interpolation in this string literal",
+                  );
```

Comment:
rustfmt gave up here, maybe it'll work after the `let else` changes, if not, please manually format.